crates/subspace-core-primitives/src/lib.rs

```diff
@@ -135,9 +135,59 @@ pub type BlockHash = [u8; 32];
 /// Slot number in Subspace network.
 pub type SlotNumber = u64;
 
+// TODO: Add related methods to `SolutionRange`.
 /// Type of solution range.
 pub type SolutionRange = u64;
 
+/// Computes the following:
+/// ```
+/// MAX * slot_probability / (pieces_in_sector * chunks / s_buckets) / sectors
+/// ```
+pub const fn sectors_to_solution_range(
+    sectors: u64,
+    slot_probability: (u64, u64),
+    max_pieces_in_sector: u16,
+) -> SolutionRange {
+    let solution_range = SolutionRange::MAX
+        // Account for slot probability
+        / slot_probability.1 * slot_probability.0
+        // Now take sector size and probability of hitting occupied s-bucket in sector into account
+        / (max_pieces_in_sector as u64 * Record::NUM_CHUNKS as u64 / Record::NUM_S_BUCKETS as u64);
+
+    // Take number of sectors into account
+    solution_range / sectors
+}
+
+/// Computes the following:
+/// ```
+/// MAX * slot_probability / (pieces_in_sector * chunks / s_buckets) / solution_range
+/// ```
+pub const fn solution_range_to_sectors(
+    solution_range: SolutionRange,
+    slot_probability: (u64, u64),
+    max_pieces_in_sector: u16,
+) -> u64 {
+    let sectors = SolutionRange::MAX
+        // Account for slot probability
+        / slot_probability.1 * slot_probability.0
+        // Now take sector size and probability of hitting occupied s-bucket in sector into account
+        / (max_pieces_in_sector as u64 * Record::NUM_CHUNKS as u64 / Record::NUM_S_BUCKETS as u64);
+
+    // Take solution range into account
+    sectors / solution_range
+}
+
+// Quick test to ensure functions above are the inverse of each other
+const_assert!(
+    solution_range_to_sectors(sectors_to_solution_range(1, (1, 1), 1000), (1, 1), 1000) == 1
+);
+const_assert!(
+    solution_range_to_sectors(sectors_to_solution_range(3, (1, 1), 1000), (1, 1), 1000) == 3
+);
+const_assert!(
+    solution_range_to_sectors(sectors_to_solution_range(5, (1, 1), 1000), (1, 1), 1000) == 5
+);
```

Comment:
The values for slot probability were `1, 6` and I'd prefer to leave them as such. Having `1, 1` can hide various edge-cases in math, so better to have realistic numbers in these assertions.

Reply:
my bad 🤦‍♂️